### ISSUE TITLE
WP admin dashboard still getting update notification after updating to 5.2.3

### ISSUE THREAD

**Issue by carl-alberto** [pantheon-systems/WordPress#216]
It seems that this last update https://github.com/pantheon-systems/WordPress/commit/25bd878d315243fc5802088bf4591022e11e4bf3 is causing persistent update nags after updating WP core 5.2.3 in the following scenarios:

- Updates done from a multidev environment in the dashboard
- Accidental updates done directly from wp-admin/update-core.php https://screencast.com/t/EZxzQu2m
- Some cases of CI/CD deploys that are happening from multidev
- Possibly updates from custom upstream

It seems this line will only check the dev environment - https://github.com/pantheon-systems/WordPress/commit/25bd878d315243fc5802088bf4591022e11e4bf3?diff=unified#diff-018e375defb06713cbd1951a7dcbb9f4R25-R27

Also, the response `https://api.live.getpantheon.com/sites/self/code-upstream-updates` does not return the response for multidevs and sometimes not reflecting the correct response from `is_up_to_date_with_upstream` the api call or getting wrong transient data

# Workarounds
These workaround may work in the dev environment only after manually overwriting core via git or sftp using this method https://pantheon.io/docs/core-updates#overwrite-core:
- Clear WP transients
- Clear Redis cache
- Terminus site:upstream:clear-cache command
- Pantheon 
- Use a previous version of the pantheon-updates.php https://github.com/pantheon-systems/WordPress/blob/5.2.2/wp-content/mu-plugins/pantheon/pantheon-updates.php if verified 5.2.3 core is installed 

Maybe for the meantime, if the users are able to see the latest wp core version from the footer after doing an update, can they safely ignore the update nag @danielbachhuber @ataylorme

**Comment by ataylorme** [comment]
The upstream update availability information is being pulled from the dashboard API, so incorrect values may be an issue there.

The WordPress transient only lasts for 2 minutes, so that shouldn't be much of an issue.

> Or, maybe for the meantime, if the users are able to see the latest wp core version from the footer after doing an update, can they safely ignore the update nag

The WordPress core update check is disabled. The previous version of the update notice would add the notice if WordPress thought core was out of date. However, that did not always correspond to when upstream updates were available.

For custom upstreams, updates are available more often than when WordPress core is out of date. For the default upstream, there is some delay between when a new version of WordPress is available and when the Pantheon upstream update is available.

Maybe as a middle ground we could check if users are on the default upstream and hide the update notice if they are on the current version of WordPress. My only concern there would be if we deploy an upstream update that doesn't correspond to a WordPress core release, such as a security of bug fix, users won't be notified. Updates to this script would be a good example.

> It seems this line will only check the dev environment - 25bd878?diff=unified#diff-018e375defb06713cbd1951a7dcbb9f4R25-R27

That seems like a valid bug. We should update `_pantheon_wordpress_update_available` to use [the `$_ENV['PANTHEON_ENVIRONMENT']` global](https://pantheon.io/docs/read-environment-config) rather than hard-coding `dev`.


cc @danielbachhuber

**Comment by ataylorme** [comment]
> Maybe as a middle ground we could check if users are on the default upstream and hide the update notice if they are on the current version of WordPress.

I submitted this in #217

**Comment by danielbachhuber** [comment]
Thanks for the report, @carl-alberto ! I appreciate the level of detail 😄

**Comment by carl-alberto** [comment]
As of WP core version 5.3.2, if still getting the error after doing the dashboard update is using this version of this file wp-content/mu-plugins/pantheon/pantheon-updates.php -> https://github.com/pantheon-systems/WordPress/blob/5.2.2/wp-content/mu-plugins/pantheon/pantheon-updates.php

**Comment by robwatson-intechra** [comment]
I'm also getting this issue repeatedly with every update. I've tried the above-mentioned workarounds but they don't result in any change. I'm loathe to hot-swap to a previous version of pantheon-updates.php. Any chance this can get fixed soon?

**Comment by carl-alberto** [comment]
Yeah, that is just a temporary fix while the engineering team is working on a long term fix regarding this issue so we do apologize if you'll have to do this on every core update :(

**Comment by greg-1-anderson** [comment]
#236 has been merged to the default branch. The default branch will be merged with master and pushed to production along with the next WordPress release per Pantheon policy to avoid intra-release update notifications on customer dashboards.
